File: app/src/main/java/com/icostel/arhitecturesample/di/modules/ActivityModule.java
package com.icostel.arhitecturesample.di.modules;

import com.icostel.arhitecturesample.ui.loginuser.LoginUserActivity;
import com.icostel.arhitecturesample.ui.main.MainActivity;
import com.icostel.arhitecturesample.ui.newuser.NewUserActivity;
import com.icostel.arhitecturesample.ui.splashscreen.SplashScreenActivity;
import com.icostel.arhitecturesample.ui.userdetails.UserDetailsActivity;

import dagger.Module;
import dagger.android.ContributesAndroidInjector;

@Module
public abstract class ActivityModule {
    // Add @ContributesAndroidInjector for each Activity we inject

    @ContributesAndroidInjector(modules = {FragmentModule.class})
    abstract LoginUserActivity contributeLoginUserActivity();

    @ContributesAndroidInjector(modules = {FragmentModule.class})
    abstract NewUserActivity contributeNewUserActivity();

    @ContributesAndroidInjector(modules = {FragmentModule.class})
    abstract SplashScreenActivity contributeSplashScreenActivity();

    @ContributesAndroidInjector(modules = {FragmentModule.class})
    abstract UserDetailsActivity contributeUserDetailsActivity();

    @ContributesAndroidInjector(modules = {FragmentModule.class})
    abstract MainActivity contributeMainActivity();
}

File: app/src/main/java/com/icostel/arhitecturesample/di/modules/ActivityModule.kt
package com.icostel.arhitecturesample.di.modules

import com.icostel.arhitecturesample.ui.screens.loginuser.LoginUserActivity
import com.icostel.arhitecturesample.ui.screens.main.MainActivity
import com.icostel.arhitecturesample.ui.screens.newuser.NewUserActivity
import com.icostel.arhitecturesample.ui.screens.splashscreen.SplashScreenActivity
import com.icostel.arhitecturesample.ui.screens.userdetails.UserDetailsActivity

import dagger.Module
import dagger.android.ContributesAndroidInjector

@Module
abstract class ActivityModule {
    // Add @ContributesAndroidInjector for each Activity we inject

    @ContributesAndroidInjector(modules = [FragmentModule::class])
    internal abstract fun contributeLoginUserActivity(): LoginUserActivity

    @ContributesAndroidInjector(modules = [FragmentModule::class])
    internal abstract fun contributeNewUserActivity(): NewUserActivity

    @ContributesAndroidInjector(modules = [FragmentModule::class])
    internal abstract fun contributeSplashScreenActivity(): SplashScreenActivity

    @ContributesAndroidInjector(modules = [FragmentModule::class])
    internal abstract fun contributeUserDetailsActivity(): UserDetailsActivity

    @ContributesAndroidInjector(modules = [FragmentModule::class])
    internal abstract fun contributeMainActivity(): MainActivity
}
